mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,487 @@
+                sk_metrics.precision_recall_curve(y, y_proba)
+            thresholds = np.append(thresholds, [1.0], axis=0)
+            pr_curve_pandas_df = pd.DataFrame(
+                {"precision": precision, "recall": recall, "thresholds": thresholds}
+            )
+            self._log_pandas_df_artifact(
+                pr_curve_pandas_df,
+                _gen_metric_name("precision_recall_curve_data"),
+            )
+
+            pr_auc = sk_metrics.auc(recall, precision)
+            self.metrics[_gen_metric_name("precision_recall_auc")] = pr_auc
+
+            def plot_pr_curve():
+                sk_metrics.PrecisionRecallDisplay(precision, recall).plot()
+
+            if hasattr(sk_metrics, 'PrecisionRecallDisplay'):
+                self._log_image_artifact(
+                    plot_pr_curve,
+                    _gen_metric_name("precision_recall_curve_plot"),
+                )
+
+    def _evaluate_classifier(self):
+        from mlflow.models.evaluation.lift_curve import plot_lift_curve
+
+        # Note: require labels to be number of 0, 1, 2, .. num_classes - 1
+        label_list = sorted(list(set(self.y)))
+        assert label_list[0] == 0, "Label values must being at '0'."
+        self.num_classes = len(label_list)
+
+        self.y_pred = self.predict_fn(self.X)
+        self.is_binomial = self.num_classes <= 2
+
+        self.metrics["accuracy"] = sk_metrics.accuracy_score(self.y, self.y_pred)
+        self.metrics["example_count"] = len(self.X)
+
+        if self.predict_proba_fn is not None:
+            self.y_probs = self.predict_proba_fn(self.X)
+            if self.is_binomial:
+                self.y_prob = self.y_probs[:, 1]
+            else:
+                self.y_prob = None
+        else:
+            self.y_probs = None
+            self.y_prob = None
+
+        if self.predict_proba_fn is not None:
+            self.metrics['log_loss'] = sk_metrics.log_loss(self.y, self.y_probs)
+
+            if self.is_binomial:
+                self._evaluate_per_class(None, self.y, self.y_pred, self.y_prob)
+                self._log_image_artifact(
+                    lambda: plot_lift_curve(self.y, self.y_probs), "lift_curve_plot",
+                )
+            else:
+                self.metrics['f1_score_micro'] = \
+                    sk_metrics.f1_score(self.y, self.y_pred, average='micro')
+                self.metrics['f1_score_macro'] = \
+                    sk_metrics.f1_score(self.y, self.y_pred, average='macro')
+                for postive_class in range(self.num_classes):
+                    y_per_class = np.where(self.y == postive_class, 1, 0)
+                    y_pred_per_class = np.where(self.y_pred == postive_class, 1, 0)
+                    pos_class_prob = self.y_probs[:, postive_class]
+                    self._evaluate_per_class(
+                        postive_class, y_per_class, y_pred_per_class, pos_class_prob
+                    )
```

Comment:
@BenWilson2 @WeichenXu123 I think users may be overwhelmed by the large number of separate metrics and artifacts here. Is there a way to condense these into fewer metrics / artifacts? One thought is that the precision / recall curves can all be included on the same plot. Per-class metrics are tougher - perhaps we should save them as a CSV table rather than as individual MLflow metrics?